src/edu/csus/ecs/pc2/clics/API202306/SubmissionService.java

```diff
@@ -607,24 +608,44 @@ public synchronized Response addNewSubmission(@Context HttpServletRequest servle
                 return Response.status(Response.Status.BAD_REQUEST).entity("no file specified").build();
             }
 
-            List<IFile> srcFiles = new ArrayList<IFile>();
-            for(CLICSFileReference file : files) {
-                String fileName = file.getFilename();
-                if("".equals(fileName)) {
-                    return Response.status(Response.Status.BAD_REQUEST).entity("no file name specified").build();
+            List<IFile> srcFiles;
+
+            // Backward compatability test: If no mime property specified on the first file, then the CLICSFileReference's are
+            // actual files and not a zip file.  This is in here because initially, due to a misinterpretation of the CLICS
+            // 2023-06 specification, the command line submit utility (a.k.a. pc2submit) did not put the files in a zip file,
+            // rather, it created an array of base64 encoded file contents.  The CLICS spec has since been clarified but for now,
+            // we will still support the incorrect implementation as well as the correct one.
+            CLICSFileReference firstFile = files[0];
+            String mimeType = firstFile.getMime();
+
+            // TODO: deprecate this "if" part and leave the "else" part.  JB
+            if(mimeType == null || "".equals(mimeType)) {
```

Comment:
A zip with an empty mime is also allowed according to the spec, I think that's not handled now?

Reply:
Makes sense!